src/main/scala/uk/gov/homeoffice/drt/analytics/prediction/FlightRouteValuesTrainer.scala

```diff
@@ -0,0 +1,151 @@
+package uk.gov.homeoffice.drt.analytics.prediction
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import akka.stream.scaladsl.{Sink, Source}
+import akka.util.Timeout
+import akka.{Done, NotUsed}
+import org.apache.spark.ml.regression.LinearRegressionModel
+import org.apache.spark.mllib.evaluation.RegressionMetrics
+import org.apache.spark.sql.types.{DoubleType, StringType, StructField, StructType}
+import org.apache.spark.sql.{Dataset, Row, SparkSession}
+import org.slf4j.LoggerFactory
+import uk.gov.homeoffice.drt.actor.PredictionModelActor.{ModelUpdate, RegressionModelFromSpark}
+import uk.gov.homeoffice.drt.actor.TerminalDateActor.FlightRoute
+import uk.gov.homeoffice.drt.analytics.prediction.FlightRouteValuesTrainer.ModelExamplesProvider
+import uk.gov.homeoffice.drt.ports.Terminals
+import uk.gov.homeoffice.drt.ports.Terminals.Terminal
+import uk.gov.homeoffice.drt.prediction.Feature.{OneToMany, Single}
+import uk.gov.homeoffice.drt.prediction.{Feature, Persistence}
+import uk.gov.homeoffice.drt.time.{SDate, SDateLike}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.CollectionConverters.seqAsJavaListConverter
+
+object FlightRouteValuesTrainer {
+  type ModelExamplesProvider[MI] = (Terminal, SDateLike, Int) => Source[(MI, Iterable[(Double, Seq[String])]), NotUsed]
+}
+
+case class FlightRouteValuesTrainer(modelName: String,
+                                    examplesProvider: ModelExamplesProvider[FlightRoute],
+                                    persistence: Persistence[FlightRoute],
+                                    baselineValue: Double,
+                                   ) {
+  private val log = LoggerFactory.getLogger(getClass)
+
+  def trainTerminals(terminals: List[Terminal])
+                    (implicit system: ActorSystem, ec: ExecutionContext, mat: Materializer, timeout: Timeout): Future[Done] =
+    Source(terminals)
+      .mapAsync(1) { terminal =>
+        train(150, 20, terminal).map(r => logStats(terminal, r))
```

Comment:
do we want 150 days to be fixed or configurable ?

Reply:
Yes you're right. Done!